airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py

```diff
@@ -24,6 +24,35 @@
 )
 
 
+@op(required_resource_keys={"all_metadata_file_blobs"})
+def remove_stale_metadata_partitions_op(context):
```

Comment:
Any reason to not perform this clean up at partition insertion time?

Reply:
Hmm I wanted to introduce this logic in a separate area right now simply because Im scared of it since its destructive.

If we accidentally delete ALL partitions, its not the end of the world, it just means we may miss a failed metadata file and have to reingest all existing metadata files to know we're ok.

Also Dagster doesn't let you bulk delete partition keys, and Im worried about the time it would take to interate over 10000 keys to delete. May lock up our sensor if deployed inside the add_partition sensor today.

So I wanted to keep it separate for now.

If things are looking good. I want to look at merging it back in.

Does that make sense?

Reply:
💯 👍 - step by step :)